src/plugins/teleport-plugin-vue-base-component/utils.ts

```diff
@@ -34,9 +34,31 @@ export const generateVueNodesTree = (
 
   if (events) {
     Object.keys(events).forEach((eventKey) => {
-      const methodName = `handle${stringToUpperCamelCase(name)}${stringToUpperCamelCase(eventKey)}`
-      methodsObject[methodName] = events[eventKey]
-      htmlUtils.addAttributeToNode(htmlNode, `@${eventKey}`, methodName)
+      const eventDetails = events[eventKey][0]
+      const isPropEvent = eventDetails && eventDetails.type === 'propCall' && eventDetails.calls
+      const isBooleanChangeEvent =
```

Comment:
you can also do one line handlers for non-boolean change events. please check the example with the tab selector, where 4 different handlers are created for the 4 buttons that are performing the tab selection

Reply:
basically, this function should be:

if - one statement - render propCall or stateChange as string
if - more statements - render eventHandlerByName, add method name to `methodsObject `

Reply:
Sure, will make the changes 👍